web_src/js/components/RepoActionView.vue

```diff
@@ -254,34 +258,38 @@ const sfc = {
           }
         }
         // append logs to the UI
-        for (const logs of job.logs.stepsLog) {
+        for (const logs of job.logs.stepsLog ?? []) {
           // save the cursor, it will be passed to backend next time
           this.currentJobStepsStates[logs.step].cursor = logs.cursor;
-          this.appendLogs(logs.step, logs.lines, logs.started);
+          this.appendLogs(logs.step, logs.started, logs.lines);
         }
 
         if (this.run.done && this.intervalID) {
           clearInterval(this.intervalID);
           this.intervalID = null;
         }
+      } catch (e) {
+        // avoid network error while unloading page, and ignore "abort" error
+        if (e instanceof TypeError || abortController.signal.aborted) return;
```

Comment:
Didn't we already have something that prevent network errors on page unload?

Also, AbortError can be detected via `e.name === 'AbortError'` as per https://stackoverflow.com/a/61741454

Reply:
No, it says clearly: `Therefore, checking abortController.signal.aborted seems like the safest.`

Reply:
> Didn't we already have something in place that prevents network errors on page unload?

I just kept the old code

Reply:
Ah I see it's just been moved. Need to check later if a generic solution for fetch is possible.